### ISSUE TITLE
UnicodeEncodeError: 'utf-8' codec can't encode characters in position 185-187: surrogates not allowed

### ISSUE THREAD

**Issue by Yaredoh** [FKNoble/path_segmentation#1]
Hi,

Thanks, for sharing it. I got this problem when i trained it. I am using Anaconda 3.6 and Tensorflow 1. 

ERROR:tornado.general:Uncaught exception, closing connection.
Traceback (most recent call last):
  File "/home/yared/anaconda3/lib/python3.6/site-packages/zmq/eventloop/zmqstream.py", line 414, in _run_callback
    callback(*args, **kwargs)
  File "/home/yared/anaconda3/lib/python3.6/site-packages/tornado/stack_context.py", line 275, in null_wrapper
    return fn(*args, **kwargs)
  File "/home/yared/anaconda3/lib/python3.6/site-packages/ipykernel/iostream.py", line 107, in _handle_event
    event_f()
  File "/home/yared/anaconda3/lib/python3.6/site-packages/ipykernel/iostream.py", line 341, in _flush
    parent=self.parent_header, ident=self.topic)
  File "/home/yared/anaconda3/lib/python3.6/site-packages/jupyter_client/session.py", line 673, in send
    to_send = self.serialize(msg, ident)
  File "/home/yared/anaconda3/lib/python3.6/site-packages/jupyter_client/session.py", line 576, in serialize
    content = self.pack(content)
  File "/home/yared/anaconda3/lib/python3.6/site-packages/jupyter_client/session.py", line 95, in <lambda>
    ensure_ascii=False, allow_nan=False,
  File "/home/yared/anaconda3/lib/python3.6/site-packages/zmq/utils/jsonapi.py", line 43, in dumps
    s = s.encode('utf8')

UnicodeEncodeError: 'utf-8' codec can't encode characters in position 185-187: surrogates not allowed

**Comment by FKNoble** [comment]
Hi there, 

I'm using Anaconda 4.3.1 with a Python 3.5 environment and TensorFlow 1.0. I've run this on a Windows 7 and Windows 10 PC and not been able to reproduce your problem. 

What operating system are you running?

**Comment by Yaredoh** [comment]
Hi, 
I am using Ubuntu 17.04 operating system, Anaconda 4.3.1 with a Python 3.6 environment and TensorFlow 1.0. The training continue for around a one minute, brings this error and stop.

**Comment by Yaredoh** [comment]
No, the result didn't show me that type of step. It generate the output file and then start some type of random symbols for around a minute. 

I believe it is from me. I used my own data set. My images are around 8gb and each images are 6000 by 6000 pixel. I made some type of modification on image size and i believe that creates encoding and decoding problem on the architecture also. 

I think, yours image data set are 224 by 224 pixel and the model you used also the same 224.

**Comment by FKNoble** [comment]
If you use the given images, does the repo' work as I've illustrated?

**Comment by Yaredoh** [comment]
It generated training_features.txt, training_labels.txt, training_record.tfrecords, validation_features.txt, validation_labels.txt, validation_record.tfrecords and the output file.... the problem is on training and visualization ...
